File: application/src/main/java/com/study/platform/domain/apply/usecase/FindAppliesUseCase.java
package com.study.platform.domain.apply.usecase;

import com.study.platform.domain.apply.dto.response.ApplyResponse;

import java.util.List;
import java.util.UUID;

public interface FindAppliesUseCase {
    List<ApplyResponse> execute(UUID userId, UUID postId);
}

File: application/src/main/kotlin/com/study/platform/domain/apply/usecase/FindAppliesUseCase.kt
package com.study.platform.domain.apply.usecase

import com.study.platform.domain.apply.dto.response.ApplyResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface FindAppliesUseCase {
    fun execute(userId: UUID, postId: UUID, pageable: Pageable): Page<ApplyResponse>
}
